internal/driver/webui.go

```diff
@@ -0,0 +1,234 @@
+	if err != nil {
+		reportHandlerError(w, "could not generate report", err)
+		return
+	}
+	g, config := report.GetDOT(rpt)
+	legend := config.Labels
+	config.Labels = nil
+	dot := &bytes.Buffer{}
+	graph.ComposeDot(dot, g, &graph.DotAttributes{}, config)
+
+	// Convert to svg.
+	svg, err := dotToSvg(dot.Bytes())
+	if err != nil {
+		reportHandlerError(w, "Failed to execute dot. Is Graphviz installed?", err)
+		return
+	}
+
+	// Get regular expression for each node.
+	nodes := []string{""}
+	for _, n := range g.Nodes {
+		nodes = append(nodes, regexp.QuoteMeta(n.Info.Name))
+	}
+
+	// Embed in html.
+	file := getFromLegend(legend, "File: ", "unknown")
+	profile := getFromLegend(legend, "Type: ", "unknown")
+	data := struct {
+		Title  string
+		Svg    template.HTML
+		Legend []string
+		Nodes  []string
+	}{
+		Title:  file + " " + profile,
+		Svg:    template.HTML(string(svg)),
+		Legend: legend,
+		Nodes:  nodes,
+	}
+	html := &bytes.Buffer{}
+	if err := graphTemplate.Execute(html, data); err != nil {
+		reportHandlerError(w, "internal template error", err)
+		return
+	}
+	w.Header().Set("Content-Type", "text/html")
+	w.Write(html.Bytes())
+}
+
+func dotToSvg(dot []byte) ([]byte, error) {
+	cmd := exec.Command("dot", "-Tsvg")
+	out := &bytes.Buffer{}
+	cmd.Stdin, cmd.Stdout, cmd.Stderr = bytes.NewBuffer(dot), out, os.Stderr
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+
+	// Fix dot bug related to unquoted amperands.
+	svg := bytes.Replace(out.Bytes(), []byte("&;"), []byte("&amp;;"), -1)
+
+	// Cleanup for embedding by dropping stuff before the <svg> start.
+	if pos := bytes.Index(svg, []byte("<svg")); pos >= 0 {
+		svg = svg[pos:]
+	}
+	return svg, nil
+}
+
+// disasm generates a web page containing disassembly.
+func (ui *webUI) disasm(w http.ResponseWriter, req *http.Request) {
+	ui.output(w, req, "disasm", "text/plain")
+}
+
+// weblist generates a web page containing disassembly.
+func (ui *webUI) weblist(w http.ResponseWriter, req *http.Request) {
+	ui.output(w, req, "weblist", "text/html")
+}
+
+// output generates a webpage that contains the output of the specified pprof cmd.
+func (ui *webUI) output(w http.ResponseWriter, req *http.Request, cmd, ctype string) {
+	focus := req.URL.Query().Get("f")
+	if focus == "" {
+		fmt.Fprintln(w, "no argument supplied for "+cmd)
+		return
+	}
+	args := []string{cmd, focus}
+	vars := pprofVariables.makeCopy()
+	_, rpt, err := generateRawReport(ui.prof, args, vars, ui.options)
+	if err != nil {
+		reportHandlerError(w, "error generating report", err)
+		return
+	}
+
+	out := &bytes.Buffer{}
+	if err := report.Generate(out, rpt, ui.options.Obj); err != nil {
+		reportHandlerError(w, "error generating report", err)
+		return
+	}
+	w.Header().Set("Content-Type", ctype)
+	io.Copy(w, out)
+}
+
+// getFromLegend returns the suffix of an entry in legend that starts
+// with param.  It returns def if no such entry is found.
+func getFromLegend(legend []string, param, def string) string {
```

Comment:
I augmented webui_test.go to check that the binary name and profile type are in the generated output.

I didn't do more than that because I was thinking that it would be best to eventually change lower-level code to produce legend info in a struct and leave the stringification to a helper routine.  In this code we would avoid the stringification altogether and just get the pieces we need directly from the struct.  E.g., I think the header we generate here could be augmented to include info about the displayed counts vs. the overall count.